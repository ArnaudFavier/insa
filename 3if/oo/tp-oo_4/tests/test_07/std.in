S segment -10 -10 -6 -10
R rectangle -8 4 4 -4
PC poly 1 1 -1 2 -3 0 -1 -3 1 0
OR union1 segment rectangle poly
LIST
MOVE segment 10 10
OR union2 union1 segment rectangle poly
LIST
HIT union1 3 4
HIT union1 -8 4
HIT union1 -6 -10
HIT union1 999 999
EXIT
