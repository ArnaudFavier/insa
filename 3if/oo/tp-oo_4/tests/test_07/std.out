OK
OK
OK
OK
poly C [(1;1) (-1;2) (-3;0) (-1;-3) (1;0)]
rectangle R [(-8;4) (4;-4)]
segment S [(-10;-10) (-6;-10)]
union1 U [S [(-10;-10) (-6;-10)] R [(-8;4) (4;-4)] C [(1;1) (-1;2) (-3;0) (-1;-3) (1;0)]]
OK
OK
poly C [(1;1) (-1;2) (-3;0) (-1;-3) (1;0)]
rectangle R [(-8;4) (4;-4)]
segment S [(0;0) (4;0)]
union1 U [S [(-10;-10) (-6;-10)] R [(-8;4) (4;-4)] C [(1;1) (-1;2) (-3;0) (-1;-3) (1;0)]]
union2 U [U [S [(-10;-10) (-6;-10)] R [(-8;4) (4;-4)] C [(1;1) (-1;2) (-3;0) (-1;-3) (1;0)]] S [(0;0) (4;0)] R [(-8;4) (4;-4)] C [(1;1) (-1;2) (-3;0) (-1;-3) (1;0)]]
YES
YES
YES
NO
